modin/pandas/series.py

```diff
@@ -1302,10 +1302,9 @@ def to_dict(self, into=dict):  # pragma: no cover
     def to_frame(self, name=None):
         from .dataframe import DataFrame
 
-        self_cp = self.copy()
         if name is not None:
-            self_cp.name = name
-        return DataFrame(self_cp)
+            name = [name]
+        return DataFrame(self, columns=name)
```

Comment:
This fix allows to use `"__reduced__"` name when casting Series to frame
```python
>>> ser.to_frame("__reduced__") # previous behaviour
   0
0  1
1  2
2  3
>>> ser.to_frame("__reduced__") # current behaviour
   __reduced__
0            1
1            2
2            3

Reply:
This is necessary since we don't want to change column names of the Series query compiler, because `by` may rely on that names